mlrun/model_monitoring/controller.py

```diff
@@ -300,17 +316,33 @@ def _should_monitor_endpoint(
                 project=endpoint.metadata.project,
                 endpoint_id=endpoint.metadata.uid,
             ) as batch_window_generator:
-                base_period_seconds = base_period_minutes * _SECONDS_IN_MINUTE
-                if application_names != batch_window_generator.get_application_list():
+                current_time = mlrun.utils.datetime_now()
+                if (
+                    # Different application names, or last analyzed never updated
+                    application_names != batch_window_generator.get_application_list()
+                    or not batch_window_generator.get_min_last_analyzed()
```

Comment:
please run `batch_window_generator.get_min_last_analyzed()` once in this method, to be sure we always work on the same one, like we are dong with the `current_time`